crates/stages/src/stages/headers.rs

```diff
@@ -313,16 +316,31 @@ where
             Ok(ExecOutput {
                 checkpoint: StageCheckpoint::new(checkpoint)
                     .with_headers_stage_checkpoint(stage_checkpoint),
-                done: true,
             })
         } else {
             Ok(ExecOutput {
                 checkpoint: current_checkpoint.with_headers_stage_checkpoint(stage_checkpoint),
-                done: false,
             })
         }
     }
 
+    async fn is_execute_done(
+        &mut self,
+        provider: &mut DatabaseProviderRW<'_, &DB>,
+        _input: ExecInput,
+        output: ExecOutput,
+    ) -> Result<bool, StageError> {
+        let gap = self
+            .get_sync_gap(
+                provider.deref(),
+                output.checkpoint.block_number,
+                // We don't mark the tip receiver as seen, so the `execute` method could do it.
+                false,
```

Comment:
My preference would be to have a separate method for this somehow as boolean flags in apis are confusing footguns